next.config.ts

```diff
@@ -46,105 +44,105 @@ const config = (phase: string, config) => {
-            use: [
+            oneOf: [
+              {
+                resourceQuery: /external/,
+                use: [{
+                  loader: 'file-loader',
+                  options: {
+                    ...staticPathConfig,
+                    name: '[name].[ext]',
+                  },
+                }],
+              },
               {
-                loader: 'url-loader',
-                options: {
-                  limit: 10000,
-                  fallback: 'file-loader',
-                  publicPath: '/_next/static/',
-                  outputPath: `${isServer ? '../' : ''}static/`,
-                  name: '[name].[ext]',
-                },
+                use: [{
+                  loader: 'url-loader',
+                  options: {
+                    ...staticPathConfig,
+                    limit: 10000,
+                    name: '[name].[ext]',
+                  },
+                }],
               },
             ],
           },
         ];
 
-        // Include all relevant directories with .tsx files that need to be transpiled
-        config.module!.rules.forEach((rule) => {
-          const ruleContainsTs = rule.test?.toString().includes('tsx');
-
-          if (ruleContainsTs) {
-            if (Array.isArray(rule.include)) {
-              rule.include = rule.include.map((includePath) => {
-                // Go down a directory to include everything from 'src'
-                if (typeof includePath === 'string' && includePath.includes('src')) {
-                  return path.resolve('src');
-                }
-
-                return includePath;
-              });
+        // Add our rules
+        if (!config.module) {
+          config.module = {
+            rules: [],
+          };
+        }
 
-              rule.include?.push(path.resolve('server'));
-            }
-          }
-        });
+        config.module.rules = [
+          ...config.module.rules!,
+          ...rules,
+        ];
 
-        // Preserve Next rules while appending our rules
-        config.module!.rules = [...config.module!.rules, ...rules];
 
         // Add plugins
-        config.plugins = config.plugins!.concat(
+        if (!config.plugins) {
+          config.plugins = [];
+        }
+
+        config.plugins = [
+          ...config.plugins,
           new webpack.DefinePlugin(GLOBALS),
           new CopyWebpackPlugin({
             patterns: [
               { from: path.resolve('public'), to: path.resolve('dist/static') },
             ],
           }),
-        );
+        ];
+
 
         // Add tsconfig paths to webpack
-        if (config.resolve) {
-          if (Array.isArray(config.resolve.plugins)) {
-            config.resolve.plugins.push(new TSConfigPathsPlugin());
-          } else {
-            config.resolve.plugins = [new TSConfigPathsPlugin()];
-          }
+        if (!config.resolve) {
+          config.resolve = {
+            plugins: [],
+          };
```

Comment:
Why is this necessary when it's empty?

Reply:
You can have a `config.resolve` object without the `plugins` property. If I would try to push something into `config.resolve.plugins` and `plugins` does not exist, the code would error. So I need to initialize `config.resolve.plugins` with an empty array if it does not exist yet.